### DISCUSSION TITLE
Safe to rename partitions?

### DISCUSSION THREAD

**Discussion by lukiz84** [TabularEditor/TabularEditor3#994] [🙋♀️TE3 Q&A] [answered]
Hi,

- SSAS Tabular on SQL Server 2019
- TabularEditor 3.10.1
- Each table has only 1 partition

Before I proceed I wanted to ask if it should be safe to run this c#-Script:

`foreach(Table t in Model.Tables) {
    foreach(Partition p in t.Partitions) {
        p.Name = t.Name + " Partition";
    }
}`

Because all of my partitions are just called "Partition" and I want them to be named according to the table they are used in.

Should I expect any problem with deployement afterwards? Because I can't make a "fresh" deployment (= delete the old cube and just deploy again)

BR
Lukas

**Comment by otykier** [comment, marked as answer]
There should not be any problems deploying the model after renaming the partitions, **HOWEVER you will lose all data in the existing partitions**, so you will need to process your model again after deployment.

**Comment by lukiz84** [comment]
Thanks Daniel - so a simple full refresh afterwards schould do it, right?

**Reply by otykier** [reply]
Correct!
